downloads/swan-lake-release-notes/swan-lake-alpha2/RELEASE_NOTE.md

```diff
@@ -2,41 +2,599 @@
+```
+
+**Table Type**
+
+Ballerina now supports configurable variables of type `table` through TOML arrays of tables.
+
+For example, if the `Config.toml` file contains the following TOML table array,
+
+```toml
+[[Pkg.users]]
+username = "alice"
+password = "password1"
+scopes = ["scope1"]
+
+[[Pkg.users]]
+username = "bob"
+password = "password2"
+scopes = ["scope1", "scope2"]
+
+[[Pkg.users]]
+username = "jack"
+password = "password3"
+```
+
+it can be loaded as a configurable variable of a `table` type as follows.
+
+```ballerina
+configurable table<AuthInfo> key(username) & readonly users = ?;
+```
+
+##### Support for Decrypting String Values Using the New Config Lang Library
+
+The `bal encrypt` command can be used to encrypt plain text values and specify those in the `Config.toml` file. Then, the `config:decryptString()` function can be used to decrypt the configurable value. 
+
+For example, if the `Config.toml` file contains the following encrypted string value,
+
+```toml
+password = "@encrypted:{ODYUoKSw0xW31eoxa/s2ESdBNgk1gX77txBIgpLC6NQ=}"
+```
+
+it can be decrypted in the Ballerina code as follows.
+
+```ballerina
+import ballerina/lang.config;
+
+configurable string password = ?;
+
+public function main() {
+    string decryptedPassword = config:decryptString(password);
+}
+```
 
 #### Standard Library
-- 
+
+##### HTTP Module Improvements
+
+###### Introduced Byte Stream Manipulation Functions to the Request and Response 
+
+This introduction enables manipulating the payload as a stream of `byte[]`. The `setByteStream()` and `getByteStream()` methods use the Ballerina stream feature.
+
+```ballerina
+http:Request request = new;
+io:ReadableByteChannel byteChannel = check io:openReadableFile("path/to/file.tmp");
+stream<io:Block, io:Error> byteStream = check byteChannel.blockStream(8196);
+request.setByteStream(byteStream);
+
+http:Response response = new;
+stream<byte[], io:Error>|error str = response.getByteStream();
+```
+
+###### Introduced the `http:Header` Annotation to Bind Headers in an Inbound Request
+
+With the introduction of the `@http:Header` annotation, inbound request headers can be retrieved by binding them to a resource method parameter. Individual headers can be accessed as `string` or `string[]` types while a parameter of type `http:Headers` can be used to access all headers. 
+
+```ballerina
+service on helloEP {
+    resource function get hello(@http:Header {name: "Accept"} string? acceptHeader, http:Headers allHeaders) {
+    //...
+    }
+}
+```
+
+##### MIME Module Improvements
+
+###### Introduced Byte Stream Manipulation Functions to the `mime:Entity` Class
+
+This introduction enables manipulating the entity body as a stream of `byte[]`.
+
+```ballerina
+function setByteStream(stream<byte[], io:Error> byteStream, string contentType = "application/octet-stream") {
+}
+
+function getByteStream(int arraySize = 8196) returns stream<byte[], io:Error>|mime:ParserError {
+}
```

Comment:
What's the purpose of these two empty functions? If the idea was to show the signature of the two methods mentioned, let's put it as a method signature: e.g.,
```ballerina
function setByteStream(stream<byte[], io:Error> byteStream, string contentType = "application/octet-stream")
```
Or maybe link to the API docs?

Reply:
@chamil321